infer/environment.cc

```diff
@@ -339,10 +339,11 @@ bool isSingleton(core::Context ctx, core::SymbolRef sym) {
 
 void Environment::updateKnowledge(core::Context ctx, core::LocalVariable local, core::Loc loc, const cfg::Send *send,
                                   KnowledgeFilter &knowledgeFilter) {
+    if (!knowledgeFilter.isNeeded(local)) {
```

Comment:
ditto

Reply:
I also believe that this is performance critical code and doing method name comparison before a hash lookup (which is hidden behind `isNeeded`) gives better perf.

Reply:
I'll move this into a separate PR.